src/openapi/streaming/streaming.js

```diff
@@ -35,35 +35,48 @@ const ignoreSubscriptions = {};
  * then follows the Connection state model.
  */
 function init() {
-    setNewContextId.call(this);
-
     this.connection = new Connection(this.options, this.baseUrl, this.transport);
 
-    updateConnectionQuery.call(this);
-
     this.connection.setStateChangedCallback(onConnectionStateChanged.bind(this));
+    this.connection.setUnauthorizedCallback(onUnauthorized.bind(this));
     this.connection.setReceivedCallback(onReceived.bind(this));
     this.connection.setConnectionSlowCallback(onConnectionSlow.bind(this));
     this.connection.setErrorCallback(onErrorCallback.bind(this));
 
     // start the connection process
-    this.connection.start(onConnectionStarted.bind(this));
+    connect.call(this);
+}
+
+/**
+ * The streaming connection received a unauthorized - the token is
+ * being rejected so we should get a new one.
+ */
+function onUnauthorized(expiryOfRejectedToken) {
+    this.authProvider.tokenRejected(expiryOfRejectedToken);
 }
 
 /**
  * Reconnects the streaming socket when it is disconnected
  */
-function reconnect() {
-    if (this.connectionState !== this.CONNECTION_STATE_DISCONNECTED) {
-        log.error(LOG_AREA, 'Only call reconnect on a disconnected streaming connection');
+function connect() {

```

Comment:
I combined connect and reconnect and made it wait for auth before trying to connect